openwisp_monitoring/db/backends/influxdb/creation.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from .. import TIMESERIES_DB
+
+logger = logging.getLogger(__name__)
+
+
+class DatabaseCreation(object):
+    ORIGINAL_DB = TIMESERIES_DB['NAME']
+    TEST_DB = f'{ORIGINAL_DB}_test'
+
+    def create_database(self, database=TIMESERIES_DB['NAME']):
+        """ creates database if necessary """
+        db = self.get_db()
+        response = db.query('SHOW DATABASES')
+        items = list(response.get_points('databases'))
+        databases = [database['name'] for database in items]
+        # if database does not exists, create it
+        if database not in databases:
+            db.create_database(database)
+            logger.info(f'Created influxdb database {database}')
+
+    def drop_database(self, database=TIMESERIES_DB['NAME']):
+        """ drops database if it exists """
+        db = self.get_db()
+        response = db.query('SHOW DATABASES')
+        items = list(response.get_points('databases'))
+        databases = [database['name'] for database in items]
+        if database in databases:
+            db.drop_database(database)
+            logger.info(f'Dropped influxdb database {database}')
+
+    def create_test_database(self):
+        TIMESERIES_DB['NAME'] = self.TEST_DB
```

Comment:
I'm not convinced about this line. I think [this approach](https://github.com/django/django/blob/master/django/db/backends/sqlite3/creation.py#L24) is much better, it's clearer and doesn't modify settings without the user noticing